src/Ztm.Zcoin.Synchronization/BlocksStorage.cs

```diff
@@ -130,14 +130,9 @@ public async Task<Block> GetAsync(int height, CancellationToken cancellationToke
                     return null;
                 }
 
-                if (data.Height == 0 && previous != null)
```

Comment:
We should keep this logic and throw when it hit so we know the data is in corrupted state.